src/libstore/build.cc

```diff
@@ -2933,8 +2933,11 @@ void DerivationGoal::runChild()
         if (drv->isBuiltin()) {
             try {
                 logger = makeJSONLogger(*logger);
-                if (drv->builder == "builtin:fetchurl")
-                    builtinFetchurl(*drv, netrcData);
+                if (drv->builder == "builtin:fetchurl") {
+                    /* fetchurl might need to rewrite its outputs, so let's pass in a closure to rewrite anythings */
+                    auto rewriter = [&](const std::string & in) { return rewriteStrings(in, inputRewrites); };
```

Comment:
The string rewriting support is all in build.cc so I didn't want to move the typedef for InputRewrites to somewhere common, when all I really needed was a `string -> string` function in `builtinFetchurl`.